### PR TITLE
Introduce thread name support

### PR DESCRIPTION
**Pull request by JorisGigxxxx**
Thank you for this great tool !

Would you please consider this trivial pull request, in order to get thread names in resulting .crash files as in the good old days... 

When available, it just extracts thread name from .ips file and add it to the thread ID line in the .crash file.

Definitely helpful in multi threaded debugging scenarios.

### CODE REVIEW

File: tool_ips2crash/ips2crash/IPSReport+CrashRepresentation.m [hunk 1]
```diff
@@ -297,9 +297,11 @@ - (NSString *)crashTextualRepresentation
         
         NSString * tCrashedString=(bThread.triggered==YES) ? @" Crashed":@"";
         
+        NSString * tNameString=(bThread.name!=nil) ? [NSString stringWithFormat:@" %@",bThread.name] : @"";
+
         NSString * tDispatchQueueString=(bThread.queue!=nil) ? [NSString stringWithFormat:@": Dispatch queue: %@",bThread.queue] : @"";
         
-        [tMutableString appendFormat:@"Thread %lu%@:%@\n",(unsigned long)bThreadIndex,tCrashedString,tDispatchQueueString];
+        [tMutableString appendFormat:@"Thread %lu%@:%@%@\n",(unsigned long)bThreadIndex,tCrashedString, tNameString, tDispatchQueueString];
```
Comment (packagesdev) [inline tool_ips2crash/ips2crash/IPSReport+CrashRepresentation.m:304]: I don't think you can have both a thread name and a dispatch queue name.
Do you have an example of a .crash or .ips file where there are both a name and a dispatch queue.

**Review by packagesdev — CHANGES_REQUESTED** [review]
This looks good and useful. The only thing that would need to be modified a bit are the changes in IPSReport+CrashRepresentation.m.

File: tool_ips2crash/ips2crash/IPSReport+CrashRepresentation.m (hunk 1 above)
Comment (JorisGigxxxx) [reply tool_ips2crash/ips2crash/IPSReport+CrashRepresentation.m:304]: Please have a look at this .ips file:
[REAPER-2022-02-11-094340.ips.txt](https://github.com/packagesdev/ips2crash/files/8047201/REAPER-2022-02-11-094340.ips.txt)

Opening this file on a Big Sur system produces this content:
`Thread 0 Crashed:: reaper Dispatch queue: com.apple.main-thread`

Here `reaper` is the thread name.

This line has been generated from this part of the .ips file:
`"name":"reaper","queue":"com.apple.main-thread","frames":[...`
 
 So I assume we can have both a name and a queue to qualify a thread.
 
 Please let me know, if you expect anything else for this contribution.

Comment (packagesdev) [reply tool_ips2crash/ips2crash/IPSReport+CrashRepresentation.m:304]: Thanks. I have searched some of my .crash archives and have found a "real" .crash log with both a thread name and a dispatch queue name.  So you're right.

I will merge the PR. I think I will need to modify the rendered string a bit after the merge as the old "real" .crash file output seems to have 2 spaces between the thread name and the dispatch queue name.

Comment (JorisGigxxxx) [reply tool_ips2crash/ips2crash/IPSReport+CrashRepresentation.m:304]: Great ! Thank you for merging this contribution. 
I reviewed the extra patch you added to match the old crash format and it looks good to me. Thanks.

**Pull request merged**